internal/access/organization.go

```diff
@@ -62,3 +62,21 @@ func DeleteOrganization(c *gin.Context, id uid.ID) error {
 func SanitizedDomain(subDomain, serverBaseDomain string) string {
 	return strings.ToLower(subDomain) + "." + serverBaseDomain
 }
+
+func UpdateOrganization(c *gin.Context, org *models.Organization) error {
+	rCtx := GetRequestContext(c)
+	if user := rCtx.Authenticated.User; user != nil && user.OrganizationID == org.ID {
+		// admins may update their own org
+		db, err := RequireInfraRole(c, models.InfraAdminRole)
+		if err != nil {
+			return HandleAuthErr(err, "organization", "update", models.InfraAdminRole)
+		}
+		return data.UpdateOrganization(db, org)
+	}
+	roles := []string{models.InfraSupportAdminRole}
```

Comment:
We can probably omit `support-admin` support in here for now and just return an error at this point